Merge295 revert

mlauster opened the pull request:
Closes #186 and #188

mlauster commented:
Replaces #295

mlauster commented on AixLib/DataBase/Boiler/General/BoilerTwoPointBaseDataDefinition.mo:
```diff
@@ -0,0 +1,35 @@
+within AixLib.DataBase.Boiler.General;
```
Is this manufacturer data publicly available? If not, do we have the permission to use it? Applies for all records....

PooyanJahangiri replied:
@AnaConstantin can you please double check this if all the data here for the boiler is publicly available? To my best knowledge all the Data I moved are available online within the manufacturers' website and they all have the source mentioned in the information tab.

mlauster replied:
@AnaConstantin, can you confirm that we can push that data to `development`?

mlauster commented on AixLib/DataBase/Boiler/WorkLoad/Buderus_GB202_40_30.mo:
```diff
@@ -0,0 +1,25 @@
+within AixLib.DataBase.Boiler.WorkLoad;
+record Buderus_GB202_40_30
+  "Buderus Logano Plus GB202 - Condensing Boiler - 40/30�C"
```
Please use only ASCII, otherwise Python cannot load model results. Please check throughout the models

mlauster commented on AixLib/DataBase/CHP/CHPBaseDataDefinition.mo:
```diff
@@ -0,0 +1,30 @@
+  parameter SI.Temperature MaxTFlow "Maximum Flow Temperature";
+  parameter SI.Temperature MaxTReturn "Maximum Return Temperature";
+
+  parameter Real Pipe_D "Outlet pipe diameter";
```
could we assign this to a unit?

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/ExternalControl_nightDayHC.mo:
```diff
@@ -0,0 +1,284 @@
+within AixLib.Fluid.HeatGenerators.BaseClasses.Controllers;
```
Could we rename the file and the model to `ExternalControlNightDayHC`?

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/ExternalControl_nightDayHC.mo:
```diff
@@ -0,0 +1,284 @@
+    "Parameters for heating curve"
+  annotation (Dialog( group= "Heating curves"), choicesAllMatching = true);
+
+  parameter Real declination = 1.1 "Declination" annotation (Dialog( group= "Heating curves"));
```
Is it helpful or necessary to have default values here? Otherwise we could just leave them out.

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/ExternalControl_nightDayHC.mo:
```diff
@@ -0,0 +1,284 @@
+  Modelica.Blocks.Sources.Constant Declination(k=declination)
+    annotation (Placement(transformation(extent={{-84,48},{-76.5,55.5}})));
+equation
+  if cardinality(isOn) < 2 then
```
If I'm not mistaken, cardinality will be removed in one of the next Modelica versions. But there are other options depending on you would like to do. We can discuss this when looking on the model together.

mlauster replied:
I don't think we should remove it and reduce usability. But if it is dependent on connector sizing, there are other options available. If I'm mistaken, then I have no immediate idea how to replace that. :) But anyhow, I would in that case keep it until a solution is available

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/InternalControl.mo:
```diff
@@ -0,0 +1,156 @@
+  parameter AixLib.DataBase.Boiler.General.BoilerTwoPointBaseDataDefinition
+    paramBoiler = AixLib.DataBase.Boiler.General.Boiler_Vitogas200F_11kW() "Parameters for Boiler"
+  annotation (Dialog(tab = "General", group = "Boiler type"), choicesAllMatching = true);
+
```
Same about default values

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/InternalControl.mo:
```diff
@@ -0,0 +1,156 @@
+            -30},{87,-2.5}})));
+equation
+
+  if cardinality(isOn) < 2 then
```
Same about cardinality

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/PITemp.mo:
```diff
@@ -0,0 +1,124 @@
+                                    annotation (Placement(transformation(
+          extent={{-120,-80},{-80,-40}}, rotation=0), iconTransformation(
+          extent={{-100,-60},{-80,-40}})));
+  Modelica.Blocks.Logical.Switch switch1
```
Please add descriptive comments

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/InternalControl.mo:
```diff
@@ -0,0 +1,156 @@
+    annotation (Placement(transformation(extent={{-13.5,6},{-4.5,15}})));
+  Modelica.Blocks.Math.Product product
+    annotation (Placement(transformation(extent={{21,24},{30,33}})));
+  Modelica.Blocks.Interfaces.RealInput Tflow_hot "outgoing Temperature [K]"
```
To real inputs and outputs, please assign units wherever applicable (by Tflow_hot(value="ThermodynamicTemperature", unit="K", displayUnit="Celsius"). You could then leave out unit definition in comment. Please check throughout all models

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/PIcontroller.mo:
```diff
@@ -0,0 +1,127 @@
+
+  Modelica.Blocks.Logical.Switch switch1
+    annotation (Placement(transformation(extent={{-70,0},{-50,20}})));
+  Modelica.Blocks.Continuous.LimPID PI(
```
Please add descriptive comments to all parameters, connectors and model instances. Please check throughout all models

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/PartialExternalControl.mo:
```diff
@@ -0,0 +1,91 @@
+        rotation=-90,
+        origin={65,-92})));
+equation
+  if cardinality(isOn) < 2 then
```
same about cardinality

mlauster commented on AixLib/Fluid/HeatGenerators/BaseClasses/Controllers/delayedOnOffController.mo:
```diff
@@ -0,0 +1,336 @@
+within AixLib.Fluid.HeatGenerators.BaseClasses.Controllers;
+model delayedOnOffController
+
```
same about default values

mlauster commented on AixLib/Fluid/HeatGenerators/Boiler.mo:
```diff
@@ -0,0 +1,132 @@
+    "Parameters for heating curve"
+  annotation (Dialog( group= "Heating curves"), choicesAllMatching = true);
+
+  parameter Real KR=1 "Gain of Boiler heater" annotation (Dialog(tab = "General", group = "Boiler type"));
```
Same about default values

mlauster commented on AixLib/Fluid/HeatGenerators/CHP.mo:
```diff
@@ -0,0 +1,268 @@
+  Modelica.Blocks.Sources.Constant ExothermicTemperature(k=1783.4)
+    annotation (Placement(transformation(extent={{44,12},{60,28}})));
+  Modelica.Blocks.Interfaces.RealInput TSet if          TSet_in
+    "Temperature setpoint [�C]" annotation (Placement(transformation(extent={{-126,76},
```
Same about ASCII characters and definition of units for connectors

mlauster commented on AixLib/Fluid/HeatGenerators/CHP.mo:
```diff
@@ -0,0 +1,268 @@
+    "Delay measurement of the controller output"
+    annotation(Dialog(group="Control system"));
+
+  BaseClasses.Controllers.delayedOnOffController delayedOnOffController(
```
Same about comments

mlauster commented on AixLib/Fluid/HeatGenerators/CHP.mo:
```diff
@@ -0,0 +1,268 @@
+        rotation=-90,
+        origin={-29.75,-89.75})));
+public
+  Modelica.Blocks.Interfaces.RealOutput ElectricalPower annotation (Placement(
```
please use small letters for instances if it's contradictive. E.g. for temperature T is fine, but Electrical should be electrical. Please check throughout the models

mlauster commented on AixLib/Fluid/HeatGenerators/CHP.mo:
```diff
@@ -0,0 +1,268 @@
+        rotation=90,
+        origin={50,90})));
+equation
+  if ElectricityDriven then
```
If this is a comparison of a parameter, it could be transferred to an initial equation or even better assigned to the parameter definition (inline)

mlauster commented on AixLib/Fluid/HeatGenerators/Examples/BoilerSystem.mo:
```diff
@@ -0,0 +1,100 @@
+within AixLib.Fluid.HeatGenerators.Examples;
+model BoilerSystem
```
same about comments

mlauster commented on AixLib/Fluid/HeatGenerators/Examples/CHPSystem.mo:
```diff
@@ -0,0 +1,61 @@
+within AixLib.Fluid.HeatGenerators.Examples;
+model CHPSystem
+  extends Modelica.Icons.Example;
```
same about comments

mlauster commented on AixLib/Fluid/HeatGenerators/Examples/HeatGeneratorNoControllSystem.mo:
```diff
@@ -0,0 +1,49 @@
+within AixLib.Fluid.HeatGenerators.Examples;
+model HeatGeneratorNoControllSystem
+  extends Modelica.Icons.Example;
```
same about comments

mlauster commented:
From my side, all review is done and the models are ready to merge! @PooyanJahangiri, could you check my changes?
@marcusfuchs, could you take responsibility for merge? I'm quite involved now in the code development, so someone else should be the gatekeeper.
Let's wait for some more days to get @AnaConstantin 's opinion on the open issue

AnaConstantin commented:
The records for the Boiler seem ok, they mention the Borschures they were taken out of so I do not see any problem there. My recomandation is to accept the pull

marcusfuchs commented:
Ok, I can do another round of Review after @PooyanJahangiri gives his ok to the current code.

PooyanJahangiri commented:
Thanks @mlauster for incorporating the necessary changes.
@marcusfuchs the code is ok, you can continue with the review.

marcusfuchs commented on AixLib/DataBase/Boiler/General/package.mo:
```diff
@@ -0,0 +1,17 @@
+
+
+
+
```
These are many empty lines.

marcusfuchs commented on AixLib/DataBase/CHP/package.mo:
```diff
@@ -0,0 +1,23 @@
+
+
+
+
```
Again, many empty lines

marcusfuchs commented on AixLib/Fluid/HeatExchangers/Examples/PumpRadiatorThermostaticValve.mo:
```diff
@@ -47,10 +47,14 @@ model PumpRadiatorThermostaticValve
-                        boiler(redeclare package Medium = Medium,
-      m_flow_nominal=0.01)     annotation(Placement(transformation(extent = {{-26, 10}, {-6, 30}})));
   Modelica.Blocks.Sources.Constant Source_TempSet_Boiler(k = 273.15 + 75) annotation(Placement(transformation(extent = {{0, 60}, {-20, 80}})));
+  AixLib.Fluid.HeatExchangers.HeaterCooler_T hea(
```
Here, I would suggest adding the change in the revision history, too.

marcusfuchs commented on AixLib/Fluid/HeatExchangers/Examples/PumpRadiatorValve.mo:
```diff
@@ -41,10 +41,14 @@ model PumpRadiatorValve
-                        boiler(redeclare package Medium = Medium,
-      m_flow_nominal=0.01)     annotation(Placement(transformation(extent = {{-26, 10}, {-6, 30}})));
   Modelica.Blocks.Sources.Constant Source_TempSet_Boiler(k = 273.15 + 75) annotation(Placement(transformation(extent = {{0, 60}, {-20, 80}})));
+  AixLib.Fluid.HeatExchangers.HeaterCooler_T hea(
```
Here, I would suggest adding the change in the revision history, too.

marcusfuchs commented on AixLib/Fluid/Storage/Examples/StorageBoiler.mo:
```diff
@@ -65,18 +61,20 @@ model StorageBoiler
     m_flow_nominal=0.5,
     dp_nominal=200)                       annotation(Placement(transformation(extent={{-28,-22},
             {-8,-2}})));
+  AixLib.Fluid.HeatExchangers.HeaterCooler_T hea(
```
Here, I would suggest adding the change in the revision history, too.

mlauster commented:
@marcusfuchs, I addressed all your remarks. @PooyanJahangiri, I added revision notes stating yours and my changes to the mentioned files. I hope that's okay for you.

marcusfuchs commented:
Perfect, thanks @PooyanJahangiri and @mlauster for the great work!
